src/main/logutils.ts

```diff
@@ -465,21 +502,11 @@ function handleChallengeModeEndLine (line: LogLine): void {
     // Calculate whether the key was timed or not
     activeChallengeMode.timed = ChallengeModeDungeon.calculateKeystoneUpgradeLevel(activeChallengeMode.allottedTime, activeChallengeMode.duration) > 0;
 
-    console.debug("[ChallengeMode] Ending current timeline segment")
-    activeChallengeMode.endCurrentTimelineSegment(videoStopDate);
+    endChallengeModeDungeon();
 
-    // If last timeline segment is less than 10 seconds long, discard it.
-    // It's probably not useful
-    const lastTimelineSegment = activeChallengeMode.getCurrentTimelineSegment();
-    if (lastTimelineSegment && lastTimelineSegment.length() < 10000) {
-        console.debug("[ChallengeMode] Removing last timeline segment, because it's too short.")
-        activeChallengeMode.removeLastTimelineSegment();
-    }
+    const result = Boolean(parseInt(line.args[1]));
 
-    console.debug("[ChallengeMode] Ending Challenge Mode instance");
-    activeChallengeMode = undefined;
-    
-    recorder.stop(metadata, overrun);
+    endRecording({result,});
```

Comment:
comma

Reply:
Also I'm curious, how does this translate to the EndRecordingOptionsType, which has 3 optional fields? Maybe the comma is required but I don't understand the syntax?

Reply:
No, the comma is not required. It's a shorthand way to specify that a variable with the same name a as a property in a type/object can be given by simply naming it.

`{result}` and `{result: result}` has the same .. result.

Optional fields are optional exactly for that reason, so we don't need to give them when we want them to have their default value as used in `endRecording()`.

Reply:
thanks for explaining!